src/Text/LLVM/PP.hs

```diff
@@ -299,34 +299,26 @@ ppTypeDecl td = ppIdent (typeName td) <+> char '='
 
 ppGlobal :: Fmt Global
 ppGlobal g = ppSymbol (globalSym g) <+> char '='
-         <+> ppTheGlobalAttrs (globalAttrs g)
+         <+> ppGlobalAttrs (isJust $ globalValue g) (globalAttrs g)
          <+> ppType (globalType g) <+> ppMaybe ppValue (globalValue g)
           <> ppAlign (globalAlign g)
           <> ppAttachedMetadata (Map.toList (globalMetadata g))
-  where
-  isStruct | Just (ValStruct {}) <- globalValue g = True
-           | otherwise = False
-  ppTheGlobalAttrs | isStruct = ppStructGlobalAttrs
-                    | otherwise = ppGlobalAttrs
 
-ppGlobalAttrs :: Fmt GlobalAttrs
-ppGlobalAttrs ga
+-- The first argument to ppGlobalAttrs indicates whether there is a value
+-- associated with this global declaration: a global declaration with a value
+-- should not be identified as "external" and "default" visibility, whereas one
```

Comment:
To avoid Haddock escaping these as module identifiers:

```suggestion
-- should not be identified as \"external\" and \"default\" visibility, whereas one
```